sseft_shap_new.py

```diff
@@ -0,0 +1,189 @@
+
+
+import shap
+import torch
+import numpy as np
+from PIL import Image
+import os, copy, sys
+import math, json
+import random
+from tqdm import tqdm
+from fairseq import checkpoint_utils, options, progress_bar, utils
+
+## This file contains the explainability framework merged with the model 'sse-ft', it is a work in progress, however the masking logic is tested and is correct
+
+## Main for now parses the args from the command prompt and passes them to load_models()
+def main():
+    print("in main")
+    parser = options.get_validation_parser()
+    args = options.parse_args_and_arch(parser)
+
+    # only override args that are explicitly given on the command line
+    override_parser = options.get_validation_parser()
+    override_args = options.parse_args_and_arch(override_parser, suppress_defaults=True)
+
+    load_models(args)
+    utils.import_user_module(args)
+    use_fp16 = args.fp16
+    use_cuda = torch.cuda.is_available() and not args.cpu
+    load_models(args)
+
+if __name__ == '__main__':
+    main()
+
+# Load_models() gets the model from the fairseq library and sets the checkpoint, it initialises a data iterator and loops over the samples in the dataset. 
+# For now I need to test the masking functions together with the model, later i will adjust the code so that it can handle data from Sound and Vision
+
+
+
+def load_models(args, override_args=None):
+    if override_args is not None:
+            overrides = vars(override_args)
+            overrides.update(eval(getattr(override_args, 'model_overrides', '{}')))
+        else:
+            overrides = None
+        ## Load ensemble and extract model
+        print('| loading model(s) from {}'.format(args.path))
```

Comment:
In the end, don't use print statements but a logger: in production, there is no terminal to write output to.